packages/dx-react-grid-bootstrap3/src/templates/column-chooser/toggle-button.jsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+export const ToggleButton = ({
+  onToggle, className,
+  getMessage, refHandler,
+  ...restProps
+}) => (
+  <button
+    className={classNames('btn btn-link', className)}
+    onClick={onToggle}
+    ref={refHandler}
+    {...restProps}
+  >
+    <i className="glyphicon glyphicon-eye-close" />
+  </button>
+);
+
+ToggleButton.propTypes = {
+  onToggle: PropTypes.func.isRequired,
+  getMessage: PropTypes.func.isRequired,
+  refHandler: PropTypes.func.isRequired,
```

Comment:
The name feels strange. It is used for the overlay target. Maybe call it get/setOverlayTarget? And make it work the same way as the getCellDimension method in TableColumnReordering?

Reply:
I'm not sure we need to point Overlay within the Button component. Because they are different components.

Reply:
Ok. Anyway, don't use the 'Handler' postfix. It is strange in react.

Reply:
It's copied from table-reordering-cell.jsx )

Reply:
But it's an internal name